tests/utils/test_readers_safedir.py

```
@@ -0,0 +1,1543 @@
+    def test_label_falls_back_when_no_file_path(self, tmp_path: Path) -> None:
+        epub_path = tmp_path / "book.epub"
+        _make_epub(epub_path)
+        with epub_path.open("rb") as f:
+            # No file_path supplied: extension check is skipped, label is <fileobj>
+            out = read_ebook_file(fileobj=f)
+        assert "Chapter one" in out
+
+    def test_rejects_non_epub_extension(self, tmp_path: Path) -> None:
+        from file_organizer.utils.readers import FileReadError as _FRE
+
+        with pytest.raises(_FRE, match="Only .epub supported"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)